File: Blind-75/151. Reverse Words in a String.py
class Solution:
    def reverseWords(self, s: str) -> str:
        n = len(s)
        word = []
        reverse_n = []
        for i in range(1,n+1):
            if s[-i] != " ":
                word.append(s[-i])
            else:
                if word:
                    word.reverse()
                    reverse_n.append(''.join(word))
                    word = [] 
        if word:
            word.reverse()
            reverse_n.append(''.join(word))
        return (' '.join(reverse_n))

File: Strings/151. Reverse Words in a String.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words;
        string temp = "";
        for(char c : s)
        {
            if(c == ' ')
            {
                if(!temp.empty())
                {
                    words.push_back(temp);
                    temp.clear();
                }
            }
            else
            {
                temp.push_back(c);
            }
        }
        if(!temp.empty())
        {
            words.push_back(temp);
        }
        reverse(words.begin(), words.end());

        string ans;
        for(string word : words)
        {
            ans += word + ' ';
        }
        if(!ans.empty())
        {
            ans.pop_back();
        }
        return ans;
    }
};
